feat: Cypress Studio - record test steps through interacting with your site

panzarino opened the pull request:
<!-- Thanks for contributing! PLEASE...
- Read our contributing guidelines: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md 
- Read our Code Review Checklist on coding standards and what needs to be done before a PR can be merged: https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md#Code-Review-Checklist
- Mark this PR as "Draft" if it is not ready for review.
-->

- Closes #73

### User facing changelog
<!-- Explain the change(s) for every user to read in our changelog. Examples: https://on.cypress.io/changelog-->

Cypress Studio - record test steps through interacting with your site

### Additional details

Users are able to add additional commands to their existing tests by clicking the new studio button. When clicked, that test will rerun and stop at its ending state. From here, users are able to interact with their site and have Cypress commands auto generated from their actions.

In addition to extending currently existing tests, users are also able to add a test to existing suites. In this case, the `before` and `beforeEach` hooks will run before tracking interactions.

**Currently supported commands**
- `cy.click()`
- `cy.type()`
- `cy.select()`
- `cy.check()`
- `cy.uncheck()`
- `cy.visit()`

Documentation - https://github.com/cypress-io/cypress-documentation/pull/3394

**Extending a Test**

![ezgif-3-73aa4f99e095](https://user-images.githubusercontent.com/7033952/104722915-7c2cf300-56fc-11eb-8b8e-640352d32f72.gif)

**Creating a New Test**

![ezgif-3-a81b254d5324](https://user-images.githubusercontent.com/7033952/104722930-7f27e380-56fc-11eb-9425-1f10c903a2e6.gif)

### PR Tasks
<!-- These tasks must be completed before a PR is merged.
Delete tasks if they are not applicable. -->

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? <!-- (internal team only)-->
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? <!-- Link to PR here -->
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [x] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

chrisbreiding commented on packages/server/test/support/fixtures/projects/studio/cypress/integration/events.spec.js:
```diff
@@ -0,0 +1,71 @@
+import { openStudio, saveStudio, verifyCommandLog } from '../support'
```
I think this spec would be a better fit as an [isolated runner spec](https://github.com/cypress-io/cypress/tree/develop/packages/runner/cypress/integration). It's easier to interact with and test the reporter/runner UI itself in those tests. The e2e snapshot here doesn't really provide any value, so we wouldn't be losing anything by getting rid of it.

In addition, it would be great to have a truly end-to-end test that interacts with Cypress Studio and then verifies that it writes the tests out to an actual file. I think that could potentially be an isolated runner test too, or a server e2e test if necessary.

panzarino replied:
@chrisbreiding yeah that was the goal here, I was having trouble figuring out how to test the file writing in a full e2e tetst so I wanted to take another stab at that today - if I can't figure it out I can transition this to somewhere else like isolated runner

chrisbreiding reviewed (changes requested):
Overall, this looks good to me. 

Some change in the CSS related to the padding on the right is causing the percy snapshot failures though, so that will have to be fixed.

panzarino commented:
@chrisbreiding I've fixed all the failing Percy snapshots - there's still one that's failing but that's intentional since I added a min-height to commands to prevent the occasional jittery effect that used to happen on hover
